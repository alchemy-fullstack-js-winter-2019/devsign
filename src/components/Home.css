.Home {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 150px 150px 150px;
  padding: 1 em;
  margin: 20px;
}
header {
  grid-column: 1 / span 4;
  background-color: #84B3DA;
}
footer {
  position: absolute;
  bottom: 0;
}
.sidebar {
  background-color: #84B3DA;
}