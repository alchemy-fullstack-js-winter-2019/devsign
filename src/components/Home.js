import React from 'react';
import UserDetails from './UserDetails';
import CreateTweetForm from './CreateTweetForm';
import Tweets from './Tweets';

const tweets = [
  {
    profileImage: 'http://cdn.onlinewebfonts.com/svg/img_191958.png',
    name: 'Ray',
    handle: '@stingraymond',
    text: 'tweet tweet tweet'
  },
  {
    profileImage: 'http://cdn.onlinewebfonts.com/svg/img_191958.png',
    name: 'Kaiya',
    handle: '@ladybeard',
    text: 'tweet tweet tweet'
  },
  {
    profileImage: 'http://cdn.onlinewebfonts.com/svg/img_191958.png',
    name: 'Randy',
    handle: '@randyroute',
    text: 'tweet tweet tweet'
  }
];

const  user = {
  name: 'Kaiya',
  handle: '@ladybeard',
  profileImage: 'http://capepremierrealty.com/wp-content/uploads/2018/11/generic-profile-icon-7.jpg.png',
  bannerImage: 'https://upload.wikimedia.org/wikipedia/commons/b/bc/Saddle_tor_to_Hey_tor_pano.jpg',
  bio: 'Fullstack MERN Developer / Animal Lover / Hiker',
  location: 'Portland, OR'
};


function Home() {
  const { name,  handle, profileImage, bannerImage, bio, location } = user;
  return (
    <main>
      <UserDetails
        name={name}
        handle={handle}
        profileImage={profileImage}
        bannerImage={bannerImage}
        bio={bio}
        location={location}
      />
      <CreateTweetForm />
      <section>
        <Tweets tweets={tweets}/>
      </section>
    </main>
  );
}

// Home.propTypes = {
//   tweets: PropTypes.array.isRequired,
//   user: PropTypes.object.isRequired
// };

export default Home;
