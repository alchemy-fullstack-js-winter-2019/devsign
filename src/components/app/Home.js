import React, { Fragment } from 'react';
import Header from './Header';
import Trending from '../chirps/Trending';
import ChirpsContainer from '../../containers/PopularChirpsContainer';
import MiniProfileContainer from '../../containers/MiniProfileContainer';
import { Div1, Div2, Section } from './HomeStyles';

export default function DesktopHome() {
  return (
    <Fragment>
      <Header />
      <Section>
        <Div1>
          <MiniProfileContainer />
          <Trending />
        </Div1>
        <Div2>
          <ChirpsContainer />
        </Div2>
      </Section>
    </Fragment>
  );
}
