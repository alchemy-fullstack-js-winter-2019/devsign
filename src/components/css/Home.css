.feed {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;

  h2 {
    color: red;
  }
}
