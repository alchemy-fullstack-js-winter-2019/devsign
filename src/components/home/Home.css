
img {
  max-height: 20vh;
  border-radius: 60px
}
ul {
  padding: 0;
}
.profile  {
  display: none;
  img {
    height: 10vh;
  }
  h3 {
    font-size: 1em;
  }
}

.post {
  margin: 2px;
  background-color: #DEEBED;
  button {
    background-color: #D8D8D8;
  }
  input {
    border: 1px solid #D8D8D8;
  }
}

