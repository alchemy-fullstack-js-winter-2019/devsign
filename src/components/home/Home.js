import React from 'react';
import styles from './Home.css';
import Tweets from '../tweets/Tweets';

const popularTweets = [
  { _id: '0001', text: 'My first tweet', user: { handle: 'Tweedy', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0002', text: 'I just flew in from Boise', user: { handle: 'Mocking Jay', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0003', text: 'A guy is sitting at an airport bar', user: { handle: 'Parrot', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0004', text: '...Tower, please call me a fuel truck.', user: { handle: 'Doug', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0005', text: 'KLM 242 expect start up in two hours', user: { handle: 'Polly', userPic: 'http://i.pravatar.cc/1000' } }
];

export default function Home() {
  return (
    <main 
      className={styles.Home}
    > 
      <Tweets tweets={popularTweets} />
    </main>
  );
}
