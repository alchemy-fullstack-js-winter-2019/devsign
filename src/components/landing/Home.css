.main {
  ul {
    padding: 0;
  }
  .profile  {
    display: none;
    img {
      height: 10vh;
    }
    h3 {
      font-size: 1em;
    }
    img {
      max-height: 20vh;
      border-radius: 60px
    }
  }
  
  .post {
    margin: 2px;
    background-color: #DEEBED;
    button {
      background-color: #D8D8D8;
    }
    input {
      border: 1px solid #D8D8D8;
    }
  }
  
  @media screen and (min-width 1000px) {
    .profile {
      
    }
  }
}  
