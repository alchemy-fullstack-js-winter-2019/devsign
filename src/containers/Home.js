import { connect } from 'react-redux';
import Home from '../components/home/Home';
import { getTweetsActions } from '../actions/Tweets';
import { getAllTweets } from '../selectors/Tweets';



const mapStateToProps = state => ({
    tweets: getAllTweets(state)
});

const mapDispatchToProps = dispatch => ({
    fetchTweets() {
        dispatch(getTweetsActions());
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Home);
