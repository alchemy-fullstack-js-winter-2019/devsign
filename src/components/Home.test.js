import React from 'react';
import Home from './Home';
import renderer from 'react-test-renderer';

describe('Home component', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <Home />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
