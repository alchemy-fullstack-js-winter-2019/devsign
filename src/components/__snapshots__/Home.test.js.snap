// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches a snapshot 1`] = `
<main>
  <aside>
    <img
      name="bannerImage"
      src="https://upload.wikimedia.org/wikipedia/commons/b/bc/Saddle_tor_to_Hey_tor_pano.jpg"
    />
    <img
      name="profileImage"
      src="http://capepremierrealty.com/wp-content/uploads/2018/11/generic-profile-icon-7.jpg.png"
    />
    <h2>
      Kaiya
       
      <span>
        @ladybeard
      </span>
    </h2>
    <p>
      Fullstack MERN Developer / Animal Lover / Hiker
    </p>
    <p>
      Portland, OR
    </p>
  </aside>
  <section>
    <form>
      <input
        name="text"
        readOnly={true}
        type="text"
      />
      <button>
        Tweet
      </button>
    </form>
  </section>
  <section>
    <section>
      <ul>
        <li>
          <article>
            <header>
              <img
                src="http://cdn.onlinewebfonts.com/svg/img_191958.png"
              />
              <h2>
                Ray
                 
                <span>
                  @stingraymond
                </span>
              </h2>
            </header>
            <p>
              tweet tweet tweet
            </p>
          </article>
        </li>
        <li>
          <article>
            <header>
              <img
                src="http://cdn.onlinewebfonts.com/svg/img_191958.png"
              />
              <h2>
                Kaiya
                 
                <span>
                  @ladybeard
                </span>
              </h2>
            </header>
            <p>
              tweet tweet tweet
            </p>
          </article>
        </li>
        <li>
          <article>
            <header>
              <img
                src="http://cdn.onlinewebfonts.com/svg/img_191958.png"
              />
              <h2>
                Randy
                 
                <span>
                  @randyroute
                </span>
              </h2>
            </header>
            <p>
              tweet tweet tweet
            </p>
          </article>
        </li>
      </ul>
    </section>
  </section>
</main>
`;
